key-share/src/valid.rs

```diff
@@ -46,13 +46,27 @@ where
             Ok(Self(T::from_parts(parts)))
         }
     }
+
+    /// Constructs `Self` from its [`ValidProjection`]
+    pub fn from<K>(value: Valid<K>) -> Self
+    where
+        T: ValidProjection<K>,
+        K: Validate,
+    {
+        Self(value.0.into())
+    }
```

Comment:
Again, I think it's better to avoid naming conflicts with `From::from` since we use those instances too. How about `from_projection`?

Reply:
Well, we don't have any `From<T>` implementations on `Valid`, so it should not lead to conflicts

Reply:
I was considering `from_proj` name, but it's uglier

Reply:
It's been removed as we have AsRef thing now